src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/StreamHelper.cs

```diff
@@ -431,8 +431,10 @@ internal static string DecodeStream(Stream stream, ref Encoding encoding)
             {
                 do
                 {
-                    // check for a charset attribute on the meta element to override the default.
-                    Match match = s_metaexp.Match(content);
+                    // check for a charset attribute on the meta element to override the default
+                    // we only look within the first 1k characters as the meta tag is in the head
+                    // tag which is at the start of the document
+                    Match match = s_metaexp.Match(content.Substring(0, 1024));
```

Comment:
Won't this throw an exception if conent.Length < 1024?   Maybe you want `content.Substring(0,Math.Min(content.Length,1024))`?

Reply:
@SteveL-MSFT Please look #16854 and Travis's comments there.
Also this could fix #16685.

Reply:
@iSazonov thanks for the pointer.  Exposing the limit as setting makes sense, but I think we can also wait til we get some user feedback that they are hitting that issue.  I'll fix the buffer overrun issue though, nice catch.